core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2659,6 +2668,13 @@ private[spark] object Utils extends Logging {
     redact(redactionPattern, kvs.toArray)
   }
 
+  def average[T](ts: Iterable[T])(implicit num: Numeric[T]): Double = {
```

Comment:
shall we require array type parameter? We are calling `size` twice in this method, which can be very slow if the input is not indexed seq.